### ISSUE TITLE
Switch to W504 from W503 (line break before a logical operator)

### ISSUE THREAD

**Issue by yelizariev** [OCA/maintainer-quality-tools#545]
Could OCA use W504, i.e.
```
if (vals.get('model') == 'ir.ui.view'
        and not self.env.context.get('duplicate_view_for_website')):
```

instead of W503, i.e.
```
if (vals.get('model') == 'ir.ui.view' and
        not self.env.context.get('duplicate_view_for_website')):
```
See
https://github.com/PyCQA/pycodestyle/issues/498
https://github.com/PyCQA/pycodestyle/pull/502/files

Feel free to add :+1: or :-1: to this post

**Comment by legalsylvain** [comment]
I personaly against such rule changes.
 
* It is matter of tastes, and doesn't provides any improvment.
* requires a lot of changes, travis will fail if we apply this change of rule.
* maintainers have other things more important than cosmetics changes like this. (IMO)

To be clear, I'm not more in favor of W504 than W503 or whatever, I just think that changing that rule will provides problems, without any improvment.

Regards.

**Comment by yelizariev** [comment]
There is always a way to apply changes softly (deactivate W503 and make W504 beta).

> It is matter of tastes, and doesn't provides any improvment.

It's actually allows read code easily. See

[1] https://mail.python.org/pipermail/python-ideas/2016-April/039774.html
[2] http://bugs.python.org/issue26763
[3] https://hg.python.org/peps/rev/3857909d7956

> maintainers have other things more important than cosmetics changes like this. (IMO)

The contributors too. Why OCA force them to update they code to W503?

**Comment by StefanRijnhart** [comment]
Agree with @legalsylvain to allow both styles.

**Comment by yelizariev** [comment]
Allowing both is also a solution

**Comment by moylop260** [comment]
Changing this style rule we will have too many travis red results.

If the current style was changed and the new one will make us headaches... then what about remove this rule?

Adding it to ignore section of the flake8 configuration file: W503

**Comment by yelizariev** [comment]
Also, we can disable W503 and apply W504 since odoo 12

**Comment by moylop260** [comment]
> Allowing both is also a solution

The technical solution is ignoring both from the configuration file of flake8.

FYI we have advanced script for pylint (to choose a special configuration file for each odoo version or PR or betas...) but for flake8 (this check is a flake8 one) we don't have this options...

**Comment by moylop260** [comment]
FYI I have taken the liberty to have created the following PR:
 - https://github.com/OCA/maintainer-quality-tools/pull/546

**Comment by pedrobaeza** [comment]
Closed by allowing both in #546
